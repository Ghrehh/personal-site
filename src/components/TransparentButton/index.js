import React from 'react';

import * as oneOrManyNodesModel from 'models/one-or-many-nodes.js';
import styles from './styles.module.css';

const TransparentButton = ({ children }) => (
  <button className={styles.button}>{children}</button>
);

TransparentButton.propTypes = {
  children: oneOrManyNodesModel.PropTypes.isRequired
};

export default TransparentButton;
