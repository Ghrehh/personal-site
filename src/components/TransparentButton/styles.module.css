.button {
  background-color: transparent;
  cursor: pointer;
  border: none;
  outline: none;
  -webkit-tap-highlight-color: transparent;
}
