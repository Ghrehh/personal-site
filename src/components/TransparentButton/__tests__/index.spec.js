import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import TransparentButton from '../';

describe('TransparentButton', () => {
  let testContext = {};

  beforeEach(() => {
    testContext = {};

    const component = (
      <TransparentButton onClick={() => {}} className="foo">
        FooBarBaz
      </TransparentButton>
    );

    testContext.renderer = new ShallowRenderer();
    testContext.renderer.render(component);
    testContext.result = testContext.renderer.getRenderOutput();
  });

  it('renders the component correctly', () => {
    expect(testContext.result).toMatchSnapshot();
  });
});
