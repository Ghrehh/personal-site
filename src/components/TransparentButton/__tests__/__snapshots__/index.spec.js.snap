// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TransparentButton renders the component correctly 1`] = `
<button
  className="button"
>
  FooBarBaz
</button>
`;
